github/downloader_test.go

```diff
@@ -151,3 +164,107 @@ func TestOnlineOrganizationDownload(t *testing.T) {
 	}
 
 }
+
+func loadReqResp(filepath string, reqResp map[string]string) error {
+	// Open a RO file
+	decodeFile, err := os.Open(filepath)
+	if err != nil {
+		return err
+	}
+	defer decodeFile.Close()
+	// Create a decoder
+	decoder := gob.NewDecoder(decodeFile)
+	// Decode
+	decoder.Decode(&reqResp)
+	return nil
+}
+
+// RoundTripFunc .
+type RoundTripFunc func(req *http.Request) *http.Response
+
+// RoundTrip .
+func (f RoundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+func TestOfflineOrganizationDownload(t *testing.T) {
+	var (
+		reqResp map[string]string = make(map[string]string)
+		err     error
+	)
+	// Load the recording
+	err = loadReqResp("../testdata/organization_src-d_2019-10-08.gob", reqResp)
+	if err != nil {
+		t.Errorf("Failed to read the offline recordings:%s", err)
+	}
+
+	// Setup the downloader with RoundTrip functionality.
+	// Not using the NewStdoutDownloader initialization because it overides the transport
+	storer := &testutils.Memory{}
+	downloader := &Downloader{
+		storer: storer,
+		client: githubv4.NewClient(&http.Client{
+			Transport: RoundTripFunc(func(req *http.Request) *http.Response {
+				// consume request body
+				savecl := req.ContentLength
+				bodyBytes, _ := ioutil.ReadAll(req.Body)
+				defer req.Body.Close()
+				// recreate request body
+				req.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+				req.ContentLength = savecl
+				data := reqResp[string(bodyBytes)]
+				return &http.Response{
+					StatusCode: 200,
+					Body:       ioutil.NopCloser(bytes.NewBufferString(data)),
+					Header:     make(http.Header),
+				}
+			})}),
+	}
+	tests, err := loadOnlineTests("../testdata/offline-organization-tests.json")
+	for _, test := range tests.OrganizationsTests {
+		t.Run(fmt.Sprintf("%s", test.Org), func(t *testing.T) {
+			testOnlineOrg(t, test, downloader, storer)
+		})
```

Comment:
if one day you decided to run these tests in parallel, it may lead to some weird bugs, please check this out https://github.com/kyoh86/scopelint

Reply:
Thanks for the tip